[IAP][UniT] Unit Testing for Storekit Implementation Features

shamin26 opened the pull request:
- Unit testing for fetching products.
Ref : https://redmine.bjitgroup.com/redmine/issues/477711

shamin26 committed:
[IAP][UniT] Unit Testing for Storekit Implementation Features

- Writing test cases by mocking product info related classes, methods.
- Test cases execution failed
- While integrating to EasyPurchase module "Product" is said to be ambiguous in the context but did not get error in demo project.

Ref : https://redmine.bjitgroup.com/redmine/issues/477111

shamin26 committed:
[IAP][UniT] Unit Testing for Storekit Implementation Features

- Writing test cases by mocking product info related classes, methods.
- Test cases executed succeessfully.

Ref : https://redmine.bjitgroup.com/redmine/issues/477111

shamin26 committed:
[IAP][UniT] Unit Testing for Storekit Implementation Features

- Writing test cases by mocking product info related classes, methods.
- Test cases executed succeessfully.
- Fixed the method name.

Ref : https://redmine.bjitgroup.com/redmine/issues/477111

sadat-399 reviewed (approved):
LGTM

TulonBJIT commented on Tests/EasyPurchaseTests/FetchProductTests.swift:
```diff
@@ -0,0 +1,69 @@
+
+// Create an extension to convert MockProduct to SKProduct
+extension MockProduct {
+    func asSKProduct() -> SKProduct {
```
Unused func. Please remove it.

shamin26 replied:
Solved.

shamin26 committed:
[IAP][UniT] Unit Testing for Storekit Implementation Features

- Writing test cases.
- Test cases executed succeessfully.
- Fixed the method name.

Ref : https://redmine.bjitgroup.com/redmine/issues/477111

TulonBJIT reviewed (approved):
Looks good to me.
